### PR TITLE
git ship: prompt when there are multiple authors

### LINKED ISSUE #486: git ship: when multiple authors, allow user to select which to set as the author
Instead of picking the author of the majority of commits, only automatically set the author if its the same for all commits, otherwise allow the user to pick which author. 

Interface could be:
* open editor to list of authors and list there commits
* user deletes the all but a single line containing the author they want to select

### PR DESCRIPTION
**Pull request by charlierudolph**
@kevgo @allewun 

resolves #486

### CODE REVIEW

File: features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    Examples:
+      | ACTION                             | FEATURE_AUTHOR                    |
+      | press ENTER                        | developer <developer@example.com> |
+      | enter "1"                          | developer <developer@example.com> |
+      | enter "2"                          | coworker <coworker@example.com>   |
+      | enter "other <other@example.com>"" | other <other@example.com>         |
+
+
+  Scenario Outline: enter invalid number
+    When I run `git ship -m 'feature done'` and enter "<NUMBER>" and "1"
```
Comment (kevgo) [inline features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature]: What is the "1" for at the end here?

File: features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    Examples:
+      | NUMBER |
+      | 0      |
+      | 3      |
+
+
+  Scenario: enter invalid custom author
+    When I run `git ship -m 'feature done'` and enter "invalid"
+    Then I get the error "Ship aborted because commit exited with error"
```
Comment (kevgo) [inline features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature]: Not a strong objection here, but is it possible to provide a more descriptive error message here, like "No author with that email found"?

File: features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    And I am left with my original commits
+
+
+  Scenario: supplying an author suppresses the prompt
```
Comment (kevgo) [inline features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature]: Scenario: supplying the author via command line arguments

File: features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature (hunk 2 above)
Comment (charlierudolph) [inline features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature]: This is really just used to tell the user to go look at the `git commit` in order to see the error. I'd rather not try and parse the git commit output.

File: features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature (hunk 1 above)
Comment (charlierudolph) [inline features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature]: The scenario outline should probably be renamed to "enter invalid number then valid number"

File: features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    When I run `git ship -m 'feature done' --author='other <other@example.com>'`
+    Then I don't see
```
Comment (kevgo) [inline features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature]: This is a very brittle spec. If we just change one part of the output, this spec will still pass. This makes TDD hard here. Could we leave this step out and the spec still makes sense?

Comment (charlierudolph) [inline features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature]: Yeah I think that would work.

File: src/helpers/git_helpers/merge_helpers.sh [hunk 5]
```diff
@@ -33,9 +33,12 @@ function commit_squash_merge {
   local branch_name=$1
   shift
   local options=$(parameters_as_string "$@")
-  local author=$(branch_author "$branch_name")
-  if [ "$author" != "$(local_author)" ]; then
-    options="--author=\"$author\" $options"
+  if ! [[ options == *"--author"* ]]; then
```
Comment (kevgo) [inline src/helpers/git_helpers/merge_helpers.sh]: What do the asterisks do here?

File: features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature (hunk 2 above)
Comment (kevgo) [inline features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature]: Makes sense. Good as is from my end.

**Comment by kevgo** [comment]
:ship: after the comments have been addressed

File: src/helpers/git_helpers/merge_helpers.sh (hunk 5 above)
Comment (charlierudolph) [inline src/helpers/git_helpers/merge_helpers.sh]: Got it from [here](http://stackoverflow.com/questions/229551/string-contains-in-bash)

File: features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+  Scenario Outline: enter invalid number
+    When I run `git ship -m 'feature done'` and enter "<NUMBER>" and "1"
+    Then I see "error: invalid number"
+    And I have the following commits
+      | BRANCH | LOCATION         | MESSAGE      | AUTHOR                            |
+      | main   | local and remote | feature done | developer <developer@example.com> |
+
+    Examples:
+      | NUMBER |
```
Comment (allewun) [inline features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature]: What about checking for letters too? Those are invalid numbers

Comment (charlierudolph) [inline features/git-ship/current_branch/on_feature_branch/without_open_changes/multiple_authors.feature]: If you don't enter a number, I'm treating the input as a custom author which I am leaving to git to validate.

**Pull request closed without merging**